### PR TITLE
FIX: Improve cutout speed

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

**Review by svandenb-dev — APPROVED** [review]
LGTM

File: src/pyedb/dotnet/edb_core/cell/primitive/primitive.py [hunk 1]
```diff
@@ -62,6 +60,14 @@ def __init__(self, pedb, edb_object):
...
         }
 
+    @property
+    def _core_stackup(self):
+        return self._app.stackup
+
+    @property
+    def _core_net(self):
+        return self._app.nets
+
```
Comment (hui-zhou-a) [inline src/pyedb/dotnet/edb_core/cell/primitive/primitive.py:70]: I suggest to remove these two properties

self._core_stackup is only used in layer_name()

self._core_net is never used.

Comment (hui-zhou-a) [inline src/pyedb/dotnet/edb_core/cell/primitive/primitive.py:70]:
```suggestion

```

Comment (hui-zhou-a) [reply src/pyedb/dotnet/edb_core/cell/primitive/primitive.py:70]: replace layer_name setter with 

```
layers = self._pedb.stackup.layers
        if isinstance(val, str) and val in layers.keys():
            layer = layers[val]._edb_layer
            if layer:
                self.primitive_object.SetLayer(layer)
            else:
                raise AttributeError("Layer {} not found.".format(val))
        elif isinstance(val, type(layers[list(layers.keys())[0]])):
            try:
                self.primitive_object.SetLayer(val._edb_layer)
            except:
                raise AttributeError("Failed to assign new layer on primitive.")
        else:
            raise AttributeError("Invalid input value")
```

Comment (maxcapodi78) [reply src/pyedb/dotnet/edb_core/cell/primitive/primitive.py:70]: I think solution 1 is more elegant. It doesn't impact on speed and there is only 1 place where the layer name is setup. Furthermore solution 2 requires much more redundant line codes

Comment (maxcapodi78) [reply src/pyedb/dotnet/edb_core/cell/primitive/primitive.py:70]: core.nets is not used in Primitive but that class is inherited in many others and there should be a need to use it

Comment (hui-zhou-a) [reply src/pyedb/dotnet/edb_core/cell/primitive/primitive.py:70]: Oh, yes. Let's keep it.

**Review by hui-zhou-a — APPROVED** [review]
LGTM.

**Pull request merged**